File: mood-tracker/frontend/src/app/shared/mood-entry.ts
export interface MoodEntry {
    id: number;
    mood: string;
    notes?: string;
    date: string;
    userId: number;
  }
  
  export interface PagedResult<T> {
    data: T[];
    page: number;
    pageSize: number;
    totalCount: number;
  }
  

File: mood-tracker/backend/MoodTracker.API/Models/MoodEntry.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MoodTracker.API.Models
{
    public class MoodEntry
    {
        public int Id { get; set; }

        [Required]
        public string Mood { get; set; } = null!; // e.g., "happy", "sad"

        public string? Notes { get; set; }

        [Required]
        public DateTime Date { get; set; } = DateTime.UtcNow;

        // Foreign Key
        public int UserId { get; set; }
    }
}
